### ISSUE TITLE
E2E tests: MaxListenersExceededWarning: Possible EventEmitter memory leak detected.

### ISSUE THREAD

**Issue by pakhuta** [nestjs/nest#4051]
## Current behavior
If in e2e tests you use `app.enableShutdownHooks();` and you have more than 10 test cases, NodeJS will print a warning: "MaxListenersExceededWarning: Possible EventEmitter memory leak detected."

## Input Code
It's just generated project by `nest-cli` without any changes except for e2e test file https://github.com/pakhuta/nestjs-max-listeners-exceeded-warning

```ts
import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { AppModule } from './../src/app.module';

describe('AppController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({ imports: [AppModule] }).compile();
    app = moduleFixture.createNestApplication();
    app.enableShutdownHooks();
    await app.init();
  });

  it.each`
      ${1}
      ${2}
      ${3}
      ${4}
      ${5}
      ${6}
      ${7}
      ${8}
      ${9}
      ${10}
      ${11}
    `('should not print warning in console', () => {
    return request(app.getHttpServer())
        .get(`/`)
        .expect(200);
  });
});

```

## Expected behavior
No warnings are printed to `stdout`.

## Possible Solution
The reason is that NestJS subscribes on list of events of process such as `SIGHUP`, `SIGINT`, etc., in `listenToShutdownSignals` method of `packages\core\nest-application-context.ts` source and doesn't unsubscribe after each test cases.

I tried this:
```ts
afterEach(async () => {
  await app.close();
});
```
but it didn't help. I think we should have some public method to shutdown application manually in `afterEach` hook.

## Environment

<pre><code>
[System Information]
OS Version     : Windows 10
NodeJS Version : v12.14.0
NPM Version    : 6.13.7

[Nest CLI]
Nest CLI Version : 6.14.2

[Nest Platform Information]
platform-express version : 6.10.14
common version           : 6.10.14
core version             : 6.10.14
</code></pre>

**Comment by kamilmysliwiec** [comment]
Fixed in 6.11.7. Remember to call `app.close()` in the `afterEach` hook

**Comment by pakhuta** [comment]
@kamilmysliwiec, thanks for quick response and fix! NestJS is awesome ❤️
